client/src/components/Auth/Register/Register.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter, Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import { registerUser } from '../../../actions/authActions';
+import { Container, Header, TopText, ContentArea, Footer } from '../Auth.styled';
+import FormGroup from '../../Shared/Forms/FormGroup';
+import { Button } from '../../Shared/Forms/Form.styled';
+
+class Register extends Component {
+	state = {
+		name: '',
+		email: '',
+		password: '',
+		password2: '',
+		errors: {},
+	};
+
+	static propTypes = {
+		registerUser: PropTypes.func.isRequired,
+		errors: PropTypes.shape({
+			name: PropTypes.string,
+			email: PropTypes.string,
+			password: PropTypes.string,
+			password2: PropTypes.string,
+		}).isRequired,
+		auth: PropTypes.shape({
+			isAuthenticated: PropTypes.bool.isRequired,
+			user: PropTypes.object.isRequired,
+		}).isRequired,
+	};
+
+	componentWillMount() {
+		const { auth, history } = this.props;
+		if (auth.isAuthenticated) history.push('/dashboard');
+	}
+
+	componentWillReceiveProps(nextProps) {
```

Comment:
This would be a great opportunity to use `getDerivedStateFromProps` instead of `componentWillReceiveProps` as this method is deprecated now. Refactor would like something like this:
```
static getDerivedStateFromProps(nextProps, prevState) {
		if (nextProps.errors) {
			return { errors: nextProps.errors };
		}
		return null;
	}
```